src/spikeinterface/core/sortinganalyzer.py

```diff
@@ -210,7 +211,7 @@ def create(
         sparsity=None,
     ):
         # some checks
-        assert sorting.sampling_frequency == recording.sampling_frequency
+        assert math.isclose(sorting.sampling_frequency, recording.sampling_frequency, abs_tol=1e-2, rel_tol=1e-5)
```

Comment:
You probably have thought deeper about this @DradeAW but what do you think of the following:

abs_tolerance of 1e-2 means that the sampling frequencies can differ by 0.01 Hz.

In other words, there is one excess cycle in the faster clock every 100 seconds.

Now, for a 30_000 Hz process this means (approximating by base) that we have a difference of  1/3 of a millisecond every 100 seconds (milliseconds_per_cycle = 1000 / 30_000 = 1/3).  In other words, 1 milliseconds every 300 seconds or 1 millisecond of difference every 5 minutes.

**That is, a relative tolerance of 0.01 Hz means 1 millisecond drift every 5 minutes.**
 
In an hour they will differ by 12 milliseconds which starts to look bad for things like template estimation, right?

Is there something wrong here in my reasoning? I am trying to think on what is the justification for the degree of leniency in this absolute tolerance.

Should the criteria be that we don't want them to differe by more than milliseconds over a day long recording? How do you think about it?

Reply:
A small difference in sampling frequency is usually the result of rounding by floating-point values, or the sorter that outputs a rounded sampling frequency ; and not an ACTUAL difference between the recording and sorting objects.

But as @samuelgarcia pointed out, it begs the question of what `analyzer.sampling_frequency` should return (as we should return the most precise sampling frequency, the one not rounded)

Reply:
My point is that with the current relative tolerance **some analysis will output incorrect results if the recording and the sorting are long enough (some hours).**.  In other words, you need some degree of long-term synch between the recording and the sorter objects and I am trying to think more sytematically about it.

Reply:
I don't understand the problem,

Both the recording and sorting use time samples (not seconds), so there is no conversion problem and they match perfectly (unless you specify spike timings in seconds, which nobody does)
I could manually set my recording sampling frequency to 30,000 and sorting to 20,000, and waveforms would still be perfect

Reply:
Actually you're right. The indices should be fine..

Reply:
if I understand well you are saying that as long as the underlying data is fine it does not matter what the `sampling_frequency` , correct?  This makes sense to me.

On top of that, you are thinking on the case where the recording and the sorter actually match but the sorter is different for any reason (like rounding). That makes sense to me as well.

Honest question:
Why do we have the check then? 
My understanding is that we are using the sampling_frequency as a proxy for synchronization of the underyling data. From that perspective, the criteria might be too lenient for the reasons described above.  I do understand that the most strict criteria throws false positives for the cases that you described.

Maybe I am wrong about the purpose of the check. What do you think?